File: android/resourceextensions/src/main/java/com/shopify/resourceextensions/LabelFactory.kt
package com.shopify.resourceextensions

class LabelFactory : Factory {

    override fun build(context: Context, args: kotlin.collections.List<Node>): Any {
        return Label(context, args)
    }

}

class Label(context: Context, args: kotlin.collections.List<Node>) {

    var caption: String? = null

    init {
        args.forEach { node ->
            (node.evaluate(context) as? Properties)?.let { properties ->
                caption = properties["caption"] as String
            }
        }
    }

}

File: ios/AppExtensionLib/AppExtensionLib/LabelFactory.swift
//
//  LabelFactory.swift
//  AppExtensionLib
//
//  Created by Dave Pelletier on 2019-10-29.
//  Copyright © 2019 Shopify. All rights reserved.
//

import Foundation

class LabelFactory : Factory {
    
    func build(context: Context, args: [Any?]) -> Any {
        var caption = ""
        args.forEach { (arg) in
            caption = (arg as? [String: Any])?["caption"] as? String ?? caption
        }
        return Label(caption)
    }
    
}

class Label {
    
    let caption: String
    
    init(_ caption: String) {
        self.caption = caption
    }
    
}
